### ISSUE TITLE
Remove Bounty Expiry from Bounties Pallet

### ISSUE THREAD

**Issue by oshakarishvili** [polkadot-fellows/runtimes#509]
The current implementation of the Bounty Expiry and Renewal feature in the Bounties pallet introduces notable inefficiencies and challenges, particularly due to its reliance on poorly designed UIs and the cumbersome processes it enforces. 

**Problem Statement:**
1. UI-Driven Challenges: The feature heavily depends on manual interactions through UI. In its current state, these interfaces are prone to issues, causing curators to miss extension calls inadvertently.
2. Inappropriate Curator Activity Signal: The expiry mechanism does not serve as a meaningful signal for curator activity. It often penalises curators unnecessarily rather than encouraging effective management of bounties.
3. Permissionless Slashing: The ability for anyone to slash curators after an expiry adds unnecessary friction. Curators then need to go through the approval process again via OpenGov, adding delays to bounty ops and redundant steps.

**Proposed Solution:**

The removal of `extend_bounty_expiry` to eliminate redundant steps and unnecessary burdens on curators, while instead relying on more effective indicators of curator activity, such as the number of bounties opened, finalised, and rewarded, reports submitted, and milestones achieved. In cases where slashing of curators is necessary, this can be addressed through OpenGov proposals, allowing token holders to collectively decide on such actions.

**References:**

1. [Parameter](https://github.com/polkadot-fellows/runtimes/blob/db4bb534cb411c0d6a2fe57eb331e6ec93ace825/relay/polkadot/src/lib.rs#L774) 
2. [Bounty Pallet](https://github.com/paritytech/polkadot-sdk/blob/1c5022014cd21cc698406684209dc88edeb2aee4/substrate/frame/bounties/src/lib.rs#L755)

**Comment by abdbee** [comment]
I agree with @oshakarishvili . Bounty expiry is never a good indicator of curatorial activities. 

It would perhaps be best to adjust the `Bounties` Pallet such that runtimes that don't want an expiry can set `BountyUpdatePeriod` to `0` and have their bounties never expire. Runtimes that want an expiry can still set the value to what they want. That way, the Bounties Pallet will serve both cases and remain backward compatible with any other runtimes using it.

**Comment by abdbee** [comment]
Another option will be to keep the expiry but refresh `update_due` for the bounty's status every time a child bounty is opened. That way, the bounty never expires as long as at least one child bounty is opened before its update period elapses.

**Comment by muharem** [comment]
we probably want to update the expire for the existing bounties as well which will require a storage migration
